origin/src/raw_mutex.rs

```diff
@@ -120,3 +133,39 @@ impl RawMutex {
         }
     }
 }
+
+pub(crate) struct Mutex<T> {
+    inner: GenericMutex<RawMutex, T>,
+}
+
+pub(crate) struct MutexGuard<'a, T> {
+    inner: GenericMutexGuard<'a, RawMutex, T>,
+}
```

Comment:
Can you comment on wrapping `lock_api::MutexGuard` in a separate `MutexGuard` type?